nilearn/image/image.py

```diff
@@ -665,3 +665,84 @@ def threshold_img(img, threshold, mask_img=None):
     threshold_img = new_img_like(img, img_data, affine)
 
     return threshold_img
+
+
+def math_img(formula, **imgs):
+    """Interpret a numpy based string formula using niimg in named parameters.
+
+    Parameters
+    ----------
+    formula: string
+        The mathematical formula to apply to image internal data.
+    imgs: images
+        Keyword arguments corresponding to the variables in the formula as
+        Nifti images. All input images should have the same geometry (shape,
+        affine).
+
+    Returns
+    -------
+    Nifti1Image
+        Result of the formula as a Nifti image. Note that the dimension of the
+        result image can be smaller than the input image. The affine is the
+        same as the input the image.
+
+    Example
+    -------
+        The same formula can be applied on different lists of Nifti images.
+
+        >>> import numpy as np
+        >>> from nibabel import Nifti1Image
+        >>> from nilearn.image import math_img
+
+        Let's create 3 sample nifti images.
+
+        >>> rng = np.random.RandomState(0)
+        >>> img1 = Nifti1Image(np.random.normal(loc=1.0,\
+                                                size=(10, 10, 10)), np.eye(4))
+        >>> img2 = Nifti1Image(np.random.normal(loc=2.0,\
+                                                size=(10, 10, 10)), np.eye(4))
+        >>> img3 = Nifti1Image(np.random.normal(loc=3.0,\
+                                                size=(10, 10, 10)), np.eye(4))
```

Comment:
That example is a bit too long for a docstring. How about using 

<pre>
from nilearn import datasets
anatomical_image = datasets.load_mni152_template()
log_img = math_img("np.log(img1)", img1=anatomical_image)
</pre>

Yes, taking the log is meaningless, but it's just to illustrate the syntax.